### ISSUE TITLE
Add support for encrypted WinRM

### ISSUE THREAD

**Issue by barakm** [xebialabs/overthere#43]
This is the default behavior for Windows VMs on EC2. 
At the moment, we have to log in to the machines, using either RDP of Powershell, and change the authentication and encryption settings.

**Comment by bpons** [comment]
Hello Vincent,

Any progress on this issue?
We'd LOVE to be able to run deployit on RHEL at BdF...
(Actually it works when setting allowunencrypted to true on WinRM, but there's no way the security is going to let us do that ;-) )

Regards,
Bernard

**Comment by radiumx3** [comment]
:+1:

**Comment by xeor** [comment]
Do anyone have any status on this issue? This issue is old, but I suspect it is still valid? Setting `allowunencrypted` isn't really an option for (I think), many people.
Is there any work-around?

**Comment by barakm** [comment]
Looking forward to this one too.
There seems to be some progress on this subject here:
https://github.com/WinRb/WinRM/pull/144

**Comment by tperrysov** [comment]
I would also love to see this.  When pushed I use an SSH client on the windows box right now, not pretty but works.

**Comment by vpartington** [comment]
Unfortunately there is no progress to report on this. Implementing support for Kerberos encryption (and for CredSSP, see #78) is very tough. The protocols are not very clearly described in the relevant Microsoft documentation.

Back in 2013 I got as far as encrypting a block with the Kerberos session key (the works is sitting on [branch kerberos-encryption](https://github.com/xebialabs/overthere/commits/kerberos-encryption)) but then got stuck because I couldn't figure out how to marshall it. Maybe the work done on the Ruby WinRM library will help here.

Unfortunately I don't know when I'll get around to fixing this myself. :-( Issue #12 has solved a lot of my use cases. But pull requests are welcome. :-)

**Comment by radiumx3** [comment]
:-1:

**Comment by davydotcom** [comment]
man I need this so bad right now too time to start digging

**Comment by davydotcom** [comment]
looks like encryption isnt too bad with the java GSS library the ruby plugin is simply using libgss

**Comment by bpons** [comment]
Tried that, unfortunately, the java gss libray is missing the extension for IOV wrapping...
see there : http://web.mit.edu/kerberos/krb5-1.12/doc/appdev/gssapi.html (IOV message wrapping)
(Could be done with JNI or JNA, i guess, but i stopped there my experiments, i'm also missing time right now to try this... )

**Comment by hierynomus** [comment]
The Ruby guys now did it.

https://github.com/WinRb/WinRM/blob/master/lib/winrm/http/transport.rb#L192
https://github.com/WinRb/WinRM/blob/master/lib/winrm/http/transport.rb#L111
https://github.com/WinRb/rubyntlm/blob/master/lib/net/ntlm/client/session.rb#L80

**Comment by digz6666** [comment]
@hierynomus But it seems like winrb doesn't support linux, you need to run on windows and run Enable-PSRemoting on powershell which doesn't yet implemented on Linux powershell.

**Comment by davydotcom** [comment]
thats not true, we use the winrm client with encryption with rubyntlm with jruby all the time. works great on linux! Would prefer to use overthere encryption instead because ummm jruby...
